### PR TITLE
test: Added a mock server for openai testing

### LINKED ISSUE #1846: Create a mock service for versioned tests
OpenAI is a http service and we will be building a mock http server that can handle a variety of calls to test out our instrumentation. We plan on using the [Python agent mock OpenAI server](https://github.com/newrelic/newrelic-python-agent/blob/develop-openai-instrumentation/tests/mlmodel_openai/_mock_external_openai_server.py) for inspiration. We will use nock to intercept the http calls and respond with a variety of responses.

**Note**: There is a [FR](https://github.com/openai/openai-node/issues/392) to replace node-fetch with undici in openai which would make our mock service useless as nock does not work.  Design this in a way that we can easily swap out what is intercepting http calls

### PR DESCRIPTION
**Pull request by jsumners-nr**
This PR adds a mock OpenAI server utilizing the mock responses in the [python agent](https://github.com/newrelic/newrelic-python-agent/blob/02ad97def2808921f102dd7dc1b5e381d4ddaa7e/tests/mlmodel_openai/_mock_external_openai_server.py#L30-L166). This resolves #1846.

### CODE REVIEW

File: test/lib/openai-mock-server.js [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const http = require('node:http')
+
+module.exports = openaiMockServer
+
+const RESPONSES = {
```
Comment (bizob2828) [inline test/lib/openai-mock-server.js:12]: might be good to move the responses to its own file

**Review by bizob2828 — APPROVED** [review]
overall this is a great start. i can see us adding more mock responses that a have full list of headers/response keys to assert full event types. Aside from moving the mocks to its own file this works as expected.

**Comment by jsumners-nr** [comment]
@bizob2828 I have refactored this PR to use a responses file in commit https://github.com/newrelic/node-newrelic/pull/1858/commits/af78c05d2ca89fc60933325bc11bdd761db95997. I opted to use a `Map` because:

1. It's easier to determine if a response exists (`.has`)
2. I find it easier to read individual responses when they are not in one massive object literal block

Note: I also fixed a mistake in the `.eslintrc.js` file.

File: .eslintrc.js [hunk 2]
```diff
@@ -26,9 +26,6 @@ module.exports = {
       }
     ]
   },
-  parserOptions: {
```
Comment (bizob2828) [inline .eslintrc.js:29]: oops, nice catch

**Review by bizob2828 — APPROVED** [review]

**Pull request merged**